docs/src/tutorial/pure.rst

```diff
@@ -209,6 +209,44 @@ Here is an example of a :keyword:`cdef` function::
         return a == b
 
 
+Managing the Global Interpreter Lock
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``cython.nogil`` can be used as a context manager or as a decorator to replace the :keyword:`nogil` keyword::
+
+    with cython.nogil:
+        # code block with the GIL released
+
+    @cython.nogil
+    @cython.cfunc
+    def func_released_gil() -> cython.int:
+        # function with the GIL released
+
+* ``cython.gil`` can be used as a context manager to replace the :keyword:`gil` keyword::
+
+    with cython.gil:
+        # code block with the GIL acquired
+
+  .. Note:: Cython currently does not support the ``@cython.gil`` decorator.
```

Comment:
I think it should be called `@cython.with_gil`. Just naming it `@cython.gil` seems unclear. (We named the modifier "... with gil" for the same reason.)

Reply:
Me sense. I will update that part.